SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultWorkerJsonGenerator.cpp

```diff
@@ -9,49 +9,60 @@
 
-bool GenerateDefaultWorkerJson(bool& bOutRedeployRequired)
+bool GenerateDefaultWorkerJson(const FString& JsonPath, FString WorkerTypeName, bool& bOutRedeployRequired)
 {
-	if (const USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetDefault<USpatialGDKEditorSettings>())
+	const FString TemplateWorkerJsonPath = FSpatialGDKServicesModule::GetSpatialGDKPluginDirectory(TEXT("SpatialGDK/Extras/templates/WorkerJsonTemplate.json"));
+
+	FString Contents;
+	if (FFileHelper::LoadFileToString(Contents, *TemplateWorkerJsonPath))
+	{
+		Contents.ReplaceInline(TEXT("{{WorkerTypeName}}"), *WorkerTypeName);
+		if (FFileHelper::SaveStringToFile(Contents, *JsonPath))
+		{
+			bOutRedeployRequired = true;
+			UE_LOG(LogSpatialGDKDefaultWorkerJsonGenerator, Verbose, TEXT("Wrote default worker json to %s"), *JsonPath)
+
+			return true;
+		}
+		else
+		{
+			UE_LOG(LogSpatialGDKDefaultWorkerJsonGenerator, Error, TEXT("Failed to write default worker json to %s"), *JsonPath)
+		}
+	}
+	else
 	{
-		const FString WorkerJsonDir = FSpatialGDKServicesModule::GetSpatialOSDirectory(TEXT("workers/unreal"));
-		const FString TemplateWorkerJsonPath = FSpatialGDKServicesModule::GetSpatialGDKPluginDirectory(TEXT("SpatialGDK/Extras/templates/WorkerJsonTemplate.json"));
+		UE_LOG(LogSpatialGDKDefaultWorkerJsonGenerator, Error, TEXT("Failed to read default worker json template at %s"), *TemplateWorkerJsonPath)
+	}
 
+	return false;
+}
+
+bool GenerateAllDefaultWorkerJsons(bool& bOutRedeployRequired)
+{
+	const FString WorkerJsonDir = FSpatialGDKServicesModule::GetSpatialOSDirectory(TEXT("workers/unreal"));
+	bool bAllJsonsGeneratedSuccessfully = true;
+
+	if (const USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetDefault<USpatialGDKEditorSettings>())
+	{
 		const FSpatialLaunchConfigDescription& LaunchConfigDescription = SpatialGDKEditorSettings->LaunchConfigDesc;
 		for (const FWorkerTypeLaunchSection& Worker : LaunchConfigDescription.ServerWorkers)
 		{
 			FString JsonPath = FPaths::Combine(WorkerJsonDir, FString::Printf(TEXT("spatialos.%s.worker.json"), *Worker.WorkerTypeName.ToString()));
 			if (!FPaths::FileExists(JsonPath))
 			{
 				UE_LOG(LogSpatialGDKDefaultWorkerJsonGenerator, Verbose, TEXT("Could not find worker json at %s"), *JsonPath);
-				FString Contents;
-				if (FFileHelper::LoadFileToString(Contents, *TemplateWorkerJsonPath))
-				{
-					Contents.ReplaceInline(TEXT("{{WorkerTypeName}}"), *Worker.WorkerTypeName.ToString());
-					if (FFileHelper::SaveStringToFile(Contents, *JsonPath))
-					{
-						bOutRedeployRequired = true;
-						UE_LOG(LogSpatialGDKDefaultWorkerJsonGenerator, Verbose, TEXT("Wrote default worker json to %s"), *JsonPath)
-					}
-					else
-					{
-						UE_LOG(LogSpatialGDKDefaultWorkerJsonGenerator, Error, TEXT("Failed to write default worker json to %s"), *JsonPath)
-					}
-				}
-				else
+
+				if (!GenerateDefaultWorkerJson(JsonPath, Worker.WorkerTypeName.ToString(), bOutRedeployRequired))
```

Comment:
Ah, this is why the param wasn't const. I'd prefer assigning this locally, and passing in a const&.

Reply:
I'm not sure I see what the problem is here. I've changed it to const and everything compiles.

Reply:
Oh - I'm surprised this compiles.